hydra_pywr/runner.py

```diff
@@ -629,16 +669,37 @@ def generate_array_recorder_resource_scenarios(self):
                     new_col_names.append(colname.split(':')[1].strip())
                 df.columns = new_col_names
 
+            if "__:" in recorder.name:
+                nodename = recorder.name.split(":")[0].replace("__", "")
+                attrname = recorder.name.split(":")[1]
+            else:
+                nodename = "network"
+                attrname = recorder.name
+
+            filename = f'{attrname}.h5'
```

Comment:
Are `attrname` (from L674) and `recorder.name` certain to be valid filenames?

Reply:
They will be valid in the sense that they will be strings which must have been parsed by pywr, will eventually represent a filename when we add the 'h5' to the end and save it in S3. Do you mean that we should try to encode this like in the previous comment to avoid unsupported characters?